config/pull_policy.go

```diff
@@ -0,0 +1,41 @@
+package config
+
+import (
+	"github.com/pkg/errors"
+)
+
+// PullPolicy defines a policy for how to manage images
+type PullPolicy int
+
+const (
+	// PullAlways images, even if they are present
+	PullAlways PullPolicy = iota
+	// PullNever images, even if they are not present
+	PullNever
+	// PullIfNotPresent pulls images if they aren't present
+	PullIfNotPresent
+)
+
+var nameMap = map[string]PullPolicy{"always": PullAlways, "never": PullNever, "if-not-present": PullIfNotPresent, "": PullAlways}
+
+// ParsePullPolicy from string
+func ParsePullPolicy(policy string) (PullPolicy, error) {
+	if val, ok := nameMap[policy]; ok {
+		return val, nil
+	}
+
+	return PullAlways, errors.Errorf("invalid pull policy %s", policy)
+}
+
+//ParsePolicyFromPull parses PullPolicy from boolean
+func ParsePolicyFromPull(pull bool) PullPolicy {
+	if pull {
+		return PullAlways
+	}
+
+	return PullNever
+}
+
+func (p PullPolicy) String() string {
+	return [...]string{"always", "never", "if-not-present"}[p]
```

Comment:
What are these hieroglyphics? 🤔

Reply:
Haha. I wanted to use these as `Enums`, but also be able to access the string form of them, so that I could assign that value to a string that would then be parsed into a `PullPolicy`. It's defining an array of strings that is accessed by the value of the `Policy` – `PullAlways` is `0`, `PullNever` is `1`, and `PullIfNotPresent` is `2` (still not implemented, btw, which is why it's still a draft)

Reply:
But I totally understand if it's not clear, and you'd want me to remove it/clarify it.

Reply:
This seems slightly error prone since it's tied to the order in which the enumeration is declared. Granted that the tests are in place to prevent any misalignment it still doesn't seem like the most ideal setup.

Reply:
Makes sense. That's why I made the tests, but it definitely doesn't seem the most secure.